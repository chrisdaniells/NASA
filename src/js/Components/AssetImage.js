import React from 'react';

export default class AssetImage extends React.Component {
    render() {
        return(
            <div className="c-asset">
                <h1 className="c-asset__title">{this.props.title}</h1>
                <p className="c-asset__description">{this.props.description}</p>
                <img src={this.props.filePath} className="c-asset__image" />
            </div>
        )
    }
}