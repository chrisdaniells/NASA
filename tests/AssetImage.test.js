import React from 'react';
import renderer from 'react-test-renderer';
import AssetImage from '../src/js/Components/AssetImage';

test('Image Asset Renders Correctly', () => {
    const component = renderer.create(
        <AssetImage 
            title="Test Title"
            description="Test Description"
            filePath="http://placehold.it/400x300"
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Audio Asset Renders Correctly', () => {
    const component = renderer.create(
        <AssetImage 
            title="Test Title"
            description="Test Description"
            filePath="http://images-assets.nasa.gov/audio/Ep7_Total%20Eclipse%20over%20America/Ep7_Total%20Eclipse%20over%20America~orig.mp3"
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});