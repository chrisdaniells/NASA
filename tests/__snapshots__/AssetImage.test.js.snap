// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Asset Should Match Snapshot 1`] = `
<div
  className="c-asset"
>
  <h1
    className="c-asset__title"
  >
    Test Title
  </h1>
  <p
    className="c-asset__description"
  >
    Test Description
  </p>
  <img
    className="c-asset__image"
    src="http://images-assets.nasa.gov/audio/Ep7_Total%20Eclipse%20over%20America/Ep7_Total%20Eclipse%20over%20America~orig.mp3"
  />
</div>
`;

exports[`Image Asset Should Match Snapshot 1`] = `
<div
  className="c-asset"
>
  <h1
    className="c-asset__title"
  >
    Test Title
  </h1>
  <p
    className="c-asset__description"
  >
    Test Description
  </p>
  <img
    className="c-asset__image"
    src="http://placehold.it/400x300"
  />
</div>
`;
